### PR TITLE
Writer refactoring

### PR DESCRIPTION
**Pull request by feuerste**
This is in preparation for #297, where we don't want to overcomplicate things.
Here we
- Merge `RawNodeWriter` and `CubeNodeWriter` into one `RawNodeWriter`, both taking an encoding enum as argument
- Move `PositionEncoding` from `octree` to `read_write`
- and a few minor changes...

### CODE REVIEW

File: src/read_write/node_iterator.rs [hunk 1]
```diff
@@ -17,25 +17,39 @@ use std::io::Result;
...
 }
 
 impl<R> NodeIterator<R>
 where
     R: NodeReader,
 {
-    pub fn new(reader: R) -> Self {
-        if reader.num_points() == 0 {
-            return NodeIterator::Empty;
+    pub fn new(reader: R, num_points: usize) -> Self {
```
Comment (nnmm) [inline src/read_write/node_iterator.rs]: Could you comment on why this changed? Does that mean you have to declare beforehand how many points you want to read?

Comment (feuerste) [reply src/read_write/node_iterator.rs]: Unfortunately this has always been there, see e.g. https://github.com/googlecartographer/point_cloud_viewer/blob/master/src/octree/generation.rs#L195 for its usage. I actually just moved it from the reader to the iterator, so at least the reader doesn't need to know it beforehand...

File: src/octree/generation.rs [hunk 2]
```diff
@@ -36,7 +37,7 @@ use std::sync::mpsc;
 const UPDATE_COUNT: i64 = 100_000;
 const MAX_POINTS_PER_NODE: i64 = 100_000;
 
-impl CubeNodeWriter {
+impl RawNodeWriter {
```
Comment (nnmm) [inline src/octree/generation.rs]: The name might not be descriptive enough now that it's not necessarily "Raw" (unencoded) anymore – how about `PerAttributeNodeWriter`?

File: src/read_write/node_iterator.rs (hunk 1 above)
Comment (nnmm) [reply src/read_write/node_iterator.rs]: Ah, okay.

File: src/octree/generation.rs (hunk 2 above)
Comment (feuerste) [reply src/octree/generation.rs]: Yeah, agree, didn't find a better name before... I meant raw in the sense of not having a file format with header... `PerAttribute...` is fine with me, too, just a little long...

**Comment by feuerste** [comment]
Thanks for the review, Nikolai!

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user feuerste. Merge queue now has a length of 1.

**Pull request merged**